src/main/java/org/opentestsystem/ap/ims/service/FileUploadServiceMapper.java

```diff
@@ -16,75 +16,95 @@
-        attachment.setFileName(brailleFile.getFileName());
-        attachment.setType(brailleFile.getType());
-        attachment.setSubtype(brailleFile.getSubtype());
-        attachment.setLanguage(brailleFile.getLanguage());
+        attachment.setFileName(attachmentFile.getFileName());
+        attachment.setType(attachmentFile.getType());
+        attachment.setSubtype(attachmentFile.getSubtype());
+        attachment.setLanguage(attachmentFile.getLanguage());
         return attachment;
     }
 
     /**
-     * Using the file name an instance of {@link BrailleFile} is created.  A file name is expected in a specific format.
+     * Using the file name an instance of {@link AttachmentFile} is created.  A file name is expected in a specific format.
      * It is assumed the file name format has been validated prior to calling this method.
      * <p>
      * <p>
      * "stim_8393_enu_contracted_transcript.brf" and "stim_8393_enu_contracted.brf" are examples of valid braille file
-     * names.  The file name contains the different parts making up a {@link BrailleFile}
+     * names. A valid asl file is "item_3431_ASL_STEM.mp4".
+     * The file name contains the different parts making up a {@link AttachmentFile}
      *
-     * @param fileName The file name to parse for the braille file data.
-     * @return A {@link BrailleFile} instance.
+     * @param fileName The file name to parse for the file data.
+     * @return A {@link AttachmentFile} instance.
      */
-    public BrailleFile newBrailleFile(final String fileName) {
+    public AttachmentFile newAttachmentFile(final String fileName, AttachmentFileType attachmentFileType) {
         if (fileName == null) {
             throw new ValidationException("File name cannot be null");
         }
 
         // first split on the extension
         final String[] splitOnExtension = split(fileName, ".");
 
-        final String brailleType = equalsIgnoreCase(splitOnExtension[1], BRAILLE_TYPE_BRF.toLowerCase()) ?
-            BRAILLE_TYPE_BRF : BRAILLE_TYPE_PRN;
+        final AttachmentFile attachmentFile = new AttachmentFile();
+        attachmentFile.setFileName(fileName);
 
+        String fileType = StringUtils.EMPTY;
         final String[] splitOnName = split(splitOnExtension[0], "_");
+        if(attachmentFileType == AttachmentFileType.BRAILLE) {
+            fileType = equalsIgnoreCase(splitOnExtension[1], BRAILLE_TYPE_BRF.toLowerCase()) ?
+                    BRAILLE_TYPE_BRF : BRAILLE_TYPE_PRN;
+
+            attachmentFile.setLanguage(splitOnName[2]);
+
+            if (splitOnName.length == 5) {
+                attachmentFile.setSubtype(splitOnName[3] + "_" + splitOnName[4]);
+            } else {
+                attachmentFile.setSubtype(splitOnName[3]);
+            }
+        } else if(attachmentFileType == AttachmentFileType.ASL) {
+            fileType = equalsIgnoreCase(splitOnExtension[1], ASL_TYPE_MP4.toLowerCase()) ?
```

Comment:
I just realized here and what I did above for braille doesn't need the "toLowerCase" because we are doing an equalsIgnoreCase.  No harm doing it I suppose.